setup.cfg

```diff
@@ -0,0 +1,18 @@
+[metadata]
+name = frequenz-channels
+author = Frequenz Energy-as-a-Service GmbH
+
+[options]
+install_requires =
+    watchfiles >= 0.15.0
+
+package_dir =
+    = src
+packages = find_namespace:
+python_requires = >= 3.8, < 4
```

Comment:
According to my experience, you need to also specify here:
```ini
include_package_data = True
```

Otherwise the `py.typed` file won't be included. You can try to build the distribution package and see if the file is included or not.

Reply:
no,  it works because we have:

``` ini
[options.package_data]
frequenz.channels = py.typed
```

Reply:
Oh, damn, it uses python package syntax in here, not distribution package syntax! It didn't work for me in another project without `include_package_data = True` but I guess then probably that config makes `setuptools` include common known data and the actual `[options.package_data]` config was simply broken. :grimacing:

Reply:
Just for completion:

`include_package_data = True` includes all the files that live on the package directory as long as:

1. These files are declared in `MANIFEST.in` (so they are part of the source distribution package)
2. OR, they are being tracked by a revision control system such as Git, Mercurial or SVN, and you have configured an appropriate plugin such as [setuptools-scm](https://pypi.org/project/setuptools-scm) or [setuptools-svn](https://pypi.org/project/setuptools-svn)

https://setuptools.pypa.io/en/latest/userguide/datafiles.html

So yeah, in our case one or the other is sufficient. I think I prefer to use `include_package_data = True` as we probably want to ship anything that we put in `src` and it is in git (otherwise it wouldn't be there). But I'm OK with both.

Reply:
Just for completion:

`include_package_data = True` includes all the files that live on the package directory as long as:

1. These files are declared in `MANIFEST.in` (so they are part of the source distribution package)
2. OR, they are being tracked by a revision control system such as Git, Mercurial or SVN, and you have configured an appropriate plugin such as [setuptools-scm](https://pypi.org/project/setuptools-scm) or [setuptools-svn](https://pypi.org/project/setuptools-svn)

https://setuptools.pypa.io/en/latest/userguide/datafiles.html

So yeah, in our case one or the other is sufficient. I think I prefer to use `include_package_data = True` as we probably want to ship anything that we put in `src` and it is in git (otherwise it wouldn't be there). But I'm OK with both.